compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7707,9 +7707,35 @@ public void visit(BLangTypeConversionExpr conversionExpr) {
     @Override
     public void visit(BLangLambdaFunction bLangLambdaFunction) {
         bLangLambdaFunction.function = rewrite(bLangLambdaFunction.function, bLangLambdaFunction.capturedClosureEnv);
+        BLangFunction function = bLangLambdaFunction.function;
+        // Add `@strand {thread: "any"}` annotation to an isolated named worker declaration in an isolated function.
+        if (function.flagSet.contains(Flag.WORKER) && Symbols.isFlagOn(function.symbol.type.flags, Flags.ISOLATED) &&
+                isInIsolatedFunction(env.enclInvokable)) {
```

Comment:
Does `isInIsolatedFunction(env.enclInvokable)` returns `true` if it is now in an inferred isolated function also? If it is so the check is okay, otherwise I think we should call a method which does that.

Reply:
Yes. The function symbol contains the `ISOLATED` flag even when it's inferred isolated since we add it in the IsolationAnayzer.